### PR TITLE
New Text plugin.

### PR DESCRIPTION
**Pull request by MiranDMC**
New Text plugin
New text comparison opcodes
Added support of condition result and strings testing to unit tests framework
New exports for listing files (future ModLoader support)
Unit tests for most of Text plugin's opcodes
Fixed string variables buffer overflows in scan_string opcode

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,7 +45,15 @@
   - new opcode **2405 ([is_script_running](https://library.sannybuilder.com/#/sa/memory/2405))**
   - new opcode **2406 ([get_script_struct_from_filename](https://library.sannybuilder.com/#/sa/memory/2406))**
   - new opcode **2407 ([is_memory_equal](https://library.sannybuilder.com/#/sa/memory/2407))**
+- new [Text](https://github.com/cleolibrary/CLEO5/tree/master/cleo_plugins/Text) plugin
+  - text related opcodes moved from CLEO core into separated plugin
+  - new opcode **2600 ([is_text_empty](https://library.sannybuilder.com/#/sa/text/2600))**
+  - new opcode **2601 ([is_text_equal](https://library.sannybuilder.com/#/sa/text/2601))**
+  - new opcode **2602 ([is_text_in_text](https://library.sannybuilder.com/#/sa/text/2602))**
+  - new opcode **2603 ([is_text_prefix](https://library.sannybuilder.com/#/sa/text/2603))**
+  - new opcode **2604 ([is_text_sufix](https://library.sannybuilder.com/#/sa/text/2604))**
```
Comment (MiranDMC) [inline CHANGELOG.md:54]:
```suggestion
  - new opcode **2604 ([is_text_suffix](https://library.sannybuilder.com/#/sa/text/2604))** 
```

File: tests/cleo_tests/cleo_tester.inc [hunk 2]
```diff
@@ -101,59 +101,173 @@ hex
...
 
+/// checks if bool value is true (different than 0)
+function assert_true(flag: int)
+    _cleo_tester_increment_assert
+	if
+		flag == false
+	then
+        _cleo_tester_fail
+        trace "TRUE expected~n~%d occured" flag
+        breakpoint
```
Comment (x87) [inline tests/cleo_tests/cleo_tester.inc:125]: maybe ?

```
breakpoint "TRUE expected~n~%d occured" flag
```

File: tests/cleo_tests/cleo_tester.inc [hunk 3]
```diff
@@ -101,59 +101,173 @@ hex
...
+end
+
+/// checks if two string values are not equal, otherwise stops the test execution
+function assert_neqs(actual:string, expected:string)
+    _cleo_tester_increment_assert
+	if
+		is_text_equal {text} actual {another} expected {ignoreCase} false
+	then
+        _cleo_tester_fail
+        trace "Expected value diferent than `%s`" actual
```
Comment (x87) [inline tests/cleo_tests/cleo_tester.inc:267]: "Expected value different than `%s`

File: tests/cleo_tests/Text/0AED.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+terminate_this_custom_script
+
+
+function tests
+    it("should print float", @test1)
+    return
+
+    function test1
+        0@v = string_float_format {number} 1.66666 {format} "Float %0.3f"
+        assert_eqs(0@v, "Float 1.667") // successfully removed
```
Comment (x87) [inline tests/cleo_tests/Text/0AED.txt:15]: nitpick: please correct the comment

File: tests/cleo_tests/cleo_tester.inc (hunk 2 above)
Comment (MiranDMC) [reply tests/cleo_tests/cleo_tester.inc:125]: breakpoint causes to put the message in the middle of log entry

File: tests/cleo_tests/Text/2600.txt [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    function test1
+        1@s = ''
+        if
+            is_text_empty 1@s
+        then
+            0@ = true
+        else
+            0@ = false
+        end
+        assert_true(0@)
```
Comment (x87) [inline tests/cleo_tests/Text/2600.txt:27]: assert_result_true ?

File: tests/cleo_tests/Text/0AED.txt (hunk 4 above)
Comment (MiranDMC) [reply tests/cleo_tests/Text/0AED.txt:15]:
```suggestion
        assert_eqs(0@v, "Float 1.667")
```

File: tests/cleo_tests/Text/2600.txt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    it("short string should NOT be empty", @test2)
+    it("long string should be empty", @test3)
+    it("long string should NOT be empty", @test4)
+    it("buffer string should be empty", @test5)
+    it("buffer string should NOT be empty", @test6)
+    return
+
+    function test1
+        1@s = ''
+        if
```
Comment (MiranDMC) [inline tests/cleo_tests/Text/2600.txt:20]:
```suggestion
        assert_result_true()
```

File: cleo_sdk/CLEO_Utils.h [hunk 7]
```diff
@@ -206,6 +211,38 @@ namespace CLEO
...
     }
 
+    struct StringParamBufferInfo
+    {
+        char* data = nullptr;
+        int size = 0;
+        BOOL needTerminator = false;
+    };
+
+    static void MemPatchJump(size_t position, void* jumpTarget)
```
Comment (x87) [inline cleo_sdk/CLEO_Utils.h:221]: can't find where it's used or moved from

File: tests/cleo_tests/cleo_tester.inc (hunk 2 above)
Comment (x87) [reply tests/cleo_tests/cleo_tester.inc:125]: sorry, what does it mean?

Comment (MiranDMC) [reply tests/cleo_tests/cleo_tester.inc:125]: Breakpoint prints multiple lines on the screen, and if I remember correctly it incorporates user's message somewhere inside the longer text ("breakpoint named X was hit"). It also changes color of the message.

File: tests/cleo_tests/Text/2600.txt (hunk 5 above)
Comment (MiranDMC) [reply tests/cleo_tests/Text/2600.txt:27]: True, it was not a thing then yet

File: cleo_sdk/CLEO_Utils.h (hunk 7 above)
Comment (MiranDMC) [reply cleo_sdk/CLEO_Utils.h:221]: https://github.com/cleolibrary/CLEO5/blob/master/source/Mem.h#L22

File: tests/cleo_tests/Text/2600.txt (hunk 5 above)
Comment (MiranDMC) [reply tests/cleo_tests/Text/2600.txt]: Fixed

**Pull request merged**